packages/frontend/src/pages/Tile/components/TableBanner/TableCollaborators.tsx

```diff
@@ -0,0 +1,169 @@
+import { ITableCollaborator, ITableCollabRole } from '@plumber/types'
+
+import { FormEventHandler, useContext, useState } from 'react'
+import { BiChevronDown, BiTrash } from 'react-icons/bi'
+import {
+  Divider,
+  Flex,
+  FormControl,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import {
+  Button,
+  ButtonProps,
+  IconButton,
+  Input,
+  Menu,
+} from '@opengovsg/design-system-react'
+import { AuthenticationContext } from 'contexts/Authentication'
+import { useTableContext } from 'pages/Tile/contexts/TableContext'
+
+const TableCollabRoleSelect = ({
+  value,
+  onChange,
+  isEditable,
+  variant = 'outline',
+}: {
+  value: ITableCollabRole
+  onChange: (val: ITableCollabRole) => void
+  isEditable: boolean
+  variant?: ButtonProps['variant']
+}) => {
+  return (
+    <Menu gutter={0}>
+      <MenuButton
+        as={Button}
+        pointerEvents={isEditable ? 'auto' : 'none'}
+        colorScheme="secondary"
+        variant={variant}
+        w={32}
+        px={6}
+        textTransform="capitalize"
+        rightIcon={isEditable ? <BiChevronDown /> : undefined}
+        textAlign={variant === 'clear' ? 'left' : 'center'}
+      >
+        {value}
+      </MenuButton>
+      <MenuList w={32}>
+        <MenuItem onClick={() => onChange('editor')}>Editor</MenuItem>
+        <MenuItem onClick={() => onChange('viewer')}>Viewer</MenuItem>
+      </MenuList>
+    </Menu>
+  )
+}
+
+const AddNewCollaborator = () => {
+  const [email, setEmail] = useState('')
+  const [role, setRole] = useState<ITableCollabRole>('editor')
+  const [isAdding, setIsAdding] = useState(false)
+
+  const onAdd: FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault()
+    setIsAdding(true)
+    console.log('Adding collaborator', email, role)
+    setIsAdding(false)
+    setEmail('')
+    setRole('editor')
+  }
+
+  return (
+    <form
+      style={{
+        width: '100%',
+      }}
+      onSubmit={onAdd}
+    >
+      <FormControl>
+        <VStack spacing={2} alignItems="flex-start">
+          <Text textStyle="subhead-3">Collaborators</Text>
+          <Flex alignSelf="stretch" gap={2}>
+            <Input
+              type="email"
+              value={email}
+              onChange={(e) => setEmail(e.target.value)}
```

Comment:
need to enforce lowercase to prevent users from whacking new emails

Reply:
i lowercased it on the backend